lib/data_update_scripts/20200410152018_resync_elasticsearch_documents.rb

```diff
@@ -0,0 +1,75 @@
+module DataUpdateScripts
+  class ResyncElasticsearchDocuments
+    def run
+      # Previous bug: Not getting removed properly
+      sync_docs(Article.pluck(:id), "Article")
+      sync_docs(Comment.pluck(:id), "Comment")
+      sync_docs(User.pluck(:id), "User")
+
+      # Previous bug: Not getting indexed on creation properly
+      index_docs(PodcastEpisode.pluck(:id), "PodcastEpisode")
+      index_docs(Tag.pluck(:id), "Tag")
+    end
+
+    private
+
+    def index_docs(ids, doc_type)
+      ids.each do |id|
+        Search::IndexToElasticsearchWorker.set(queue: :low_priority).perform_async(
+          doc_type, id
+        )
+      end
+    end
+
+    # This is essentially doing a "find_each" in Elasticsearch by scrolling through
+    # all of the documents and collecting their IDs. Then we remove the mismatched Postgres ones
+    def sync_docs(mysql_ids, doc_type)
+      es_ids = []
+      response = Search::Client.search(
+        index: search_class(doc_type)::INDEX_ALIAS, scroll: "2m", body: search_body(doc_type), size: 3000,
```

Comment:
what does "2m" as a value mean?

Reply:
2 minutes, its honestly way longer than we probably need in this case but basically what is means is that the time it takes to grab the records and process them needs to be less than 2 minutes, after that then the scroll will time out. It would be like putting a time limit on `find_each` for processing each batch.